### ISSUE TITLE
PT: Remove Customizer Menu Item for Block Themes

### ISSUE THREAD

**Issue by dsas** [Automattic/wp-calypso#78938]
**PT: paYJgx-3CN-p2**

### Details

We [removed the Additional CSS menu item from the menu](https://github.com/Automattic/wp-calypso/issues/71130) and now it's time to build upon that and remove Customizer from the menu for sites that are using block themes and are not using a theme or plugin which requires use of the customiser.

We can do this by ensuring that the Customizer link is added conditionally - following the [core logic](https://github.com/WordPress/WordPress/blob/bbfe308f3335ab1bf10a60720980c7b87ca9f4b1/wp-admin/menu.php#L219), rather than [always being added](https://github.com/Automattic/jetpack/blob/a3dfba95f4b31899fe042ab5f8da95b41f2cb2fe/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php#L277).

The Customizer is still needed for a bunch of features such as: [Social links](https://github.com/Automattic/jetpack/blob/2e69b7fe9f9eba8496932296fe022f698ccc1e8a/projects/plugins/jetpack/modules/theme-tools/social-links.php#L134), [Jetpack Search](https://github.com/Automattic/jetpack/blob/2e69b7fe9f9eba8496932296fe022f698ccc1e8a/projects/packages/search/src/customizer/class-customizer.php#L38), [Testimonials](https://github.com/Automattic/jetpack/blob/2e69b7fe9f9eba8496932296fe022f698ccc1e8a/projects/plugins/jetpack/modules/custom-post-types/testimonial.php#L532), [Related posts](https://github.com/Automattic/jetpack/blob/2e69b7fe9f9eba8496932296fe022f698ccc1e8a/projects/plugins/jetpack/modules/related-posts/class.related-posts-customize.php#L48), [Featured content](https://github.com/Automattic/jetpack/blob/2e69b7fe9f9eba8496932296fe022f698ccc1e8a/projects/plugins/jetpack/modules/theme-tools/featured-content.php#L511), [Portfolios](https://github.com/Automattic/jetpack/blob/2e69b7fe9f9eba8496932296fe022f698ccc1e8a/projects/plugins/jetpack/modules/custom-post-types/portfolios.php#L83), [Footer credit](https://wordpress.com/support/footer-credits/), [WooCommerce](https://woocommerce.com/document/woocommerce-customizer/), however, except footer-credits none of these are enabled by default but are behind settings or by buying upgrades (e.g. jetpack search). T-Rex should document these and ping the relevant owners. The people who maintain that code will be best placed to investigate the customizer use and change it as appropriate.

It seems likely that the footer-credits code is unowned, and given it still relies upon the customizer and forces the menu to be appeared, it might need T-Rex to action that.

See discussion: p1688458287536349-slack-C029GN3KD
It contributes towards issues raised in the May NMIT (p58i-evH-p2) around providing two interfaces for editing menus.
```[tasklist]
### Tasks ([see details](https://github.com/Automattic/wp-calypso/issues/78938#issuecomment-1633656692))
- [ ] https://github.com/Automattic/dotcom-forge/issues/3260
- [ ] Headstart `wpcom/wp-content/lib/headstart/init.php?r=ec665df1#134`
- [ ] Site Logo `wpcom/wp-content/mu-plugins/avatars.php?r=c7ebb12e#816`
- [ ] Footer Credits `wpcom/wp-content/mu-plugins/footer-credit/customizer.php?r=7400eefe#117`
- [x] Jetpack Social Links
- [ ] Logo Maker Upsell `wpcom/wp-content/mu-plugins/logo-tool/logo-tool.php?r=7d66e22c#81`
- [ ] Jetpack Fonts `wpcom/wp-content/plugins/custom-fonts/custom-fonts.php?r=173f53e3#97-98`
- [ ] Typekit Fonts `wpcom/wp-content/plugins/custom-fonts-typekit/custom-fonts-typekit.php?r=884bf172#50`
- [ ] Atomic: Custom CSS (Free Sites)
- [ ] Atomic: Footer Credits (Free Sites)
- [ ] Atomic: Create Logo
- [ ] Atomic: Jetpack Fonts
- [ ] Atomic: Typekit Fonts
- [ ] Custom CSS nudge `jetpack/projects/plugins/jetpack/modules/masterbar/nudges/bootstrap.php?r=3a3c4adc#60`
```

**Comment by mmtr** [comment]
> We can do this by ensuring that the Customizer link is added conditionally - following the [core logic](https://github.com/WordPress/WordPress/blob/bbfe308f3335ab1bf10a60720980c7b87ca9f4b1/wp-admin/menu.php#L219), rather than [always being added](https://github.com/Automattic/jetpack/blob/a3dfba95f4b31899fe042ab5f8da95b41f2cb2fe/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php#L277).

We are already doing that, so I'm afraid that removing the Customizer menu is not that trivial. If the menu is showing up in Simple and Atomic sites by default, is because there is some code registering a `customize_register` action which doesn't require any additional purchase or a manual setting activation.

**Comment by dsas** [comment]
My apologies, I'd misunderstood the nav-unification logic 😞 That just makes the task a little easier though - as we don't need to change nav-unification.

There are manylots calls to hook `customize_register` which is what we and dotorg use to decide whether to display the Customizer link. I've audited all of them to see which need addressing. I used a simple, free site with a block theme enabled for this - it seems likely that all/most calls that affect free will probably affect paid plans.

There are 8 calls to `customize_register` across seven files that cause the Customizer menu link to show up. These are:

- [ ] Headstart - Possibly not used on block themes but needs further research fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Syvo%2Surnqfgneg%2Svavg.cuc%3Se%3Qrp665qs1%23134-og
- [ ] Site Logo - Removes the site logo control if a blavatar is assigned, part of a six year-old deprecation, see D4229-code - fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Sningnef.cuc%3Se%3Qp7roo12r%23816-og
- [ ] Footer credits - will need an alternative developed - fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Ssbbgre%2Qperqvg%2Sphfgbzvmre.cuc%3Se%3Q7400rrsr%23117-og
- [x] Jetpack Social Links - will be fixed by https://github.com/Automattic/jetpack/pull/31838
- [ ] Logo maker upsell - p5uIfZ-cPi-p2 appears to miss it but it doesn't work on TT3, TT2 or Hexa (classic) as far as I can tell, looks like the theme needs to opt-in - fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Sybtb%2Qgbby%2Sybtb%2Qgbby.cuc%3Se%3Q7q66r22p%2381-og
- [ ] Jetpack fonts (x2 calls) - unsure what's going on - I don't get fonts in customizer with TT3 at least - fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Scyhtvaf%2Sphfgbz%2Qsbagf%2Sphfgbz%2Qsbagf.cuc%3Se%3Q173s53r3%2397-og
- [ ] Typekit fonts - unsure what's going on - I don't get fonts in customizer with TT3 at least - fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Scyhtvaf%2Sphfgbz%2Qsbagf%2Qglcrxvg%2Sphfgbz%2Qsbagf%2Qglcrxvg.cuc%3Se%3Q884os172%2350-og

**Comment by simison** [comment]
Thanks! I'll respond with what I know about each:

> Headstart

Runs only when new sites are created, right?

> Site logo

Is this the WP core setting? Core doesn't show customizer either so I'd follow what they do for this. We have site logo setting in Calypso.

> Logo maker upsell 

We have the upsell in Calypso Settings exactly so that we don't lose sales from customizer upselll; ok to hide.

> Jetpack fonts

A feature for classic themes; block themes use Jetpack fonts via global styles.

> Typekit fonts

Typekit fonts were deprecated like 5+ years ago I think, so safe to not deal with block themes and TypeKit customizer.

_Footer credits_ seems like the only real issue here.

**Comment by mmtr** [comment]
For Atomic sites I also found that the `custom-css` module is enabled by default (fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Syvo%2Sngbzvp%2Spynff%2Qwrgcnpx%2Qznantre.cuc%3Se%3Q0ppqr343%2328-og) which always [hooks into the `customize_register` action](https://github.com/Automattic/jetpack/blob/3fb51552fc8215fbe87cca93499e1415078fb5dc/projects/plugins/jetpack/modules/custom-css/custom-css.php#L26) effectively making the Customize menu to be always visible (unless users disable the `custom-css` module).

We should probably keep the module disabled during the AT process if the site has a block theme active.

**Comment by mmtr** [comment]
More Atomic cases that use the `customize_register` action:
- Atomic Free sites to show a nudge in the Customizer that upsells the Custom CSS feature:
  - https://github.com/Automattic/jetpack/blob/3fb51552fc8215fbe87cca93499e1415078fb5dc/projects/plugins/jetpack/modules/masterbar/nudges/bootstrap.php#L55
  - https://github.com/Automattic/wpcomsh/blob/2228359875ed6f13ae1164358457971dcdaa863d/feature-plugins/additional-css.php#L29)
- Atomic Free sites to show a nudge in the Customizer that upsells the footer credit removal:
  - https://github.com/Automattic/wpcomsh/blob/2228359875ed6f13ae1164358457971dcdaa863d/footer-credit/footer-credit/customizer.php#L113
- "Create logo" button in the Customizer:
  - https://github.com/Automattic/wpcomsh/blob/2228359875ed6f13ae1164358457971dcdaa863d/logo-tool/logo-tool.php#L89
- Jetpack / Typekit fonts also affect Atomic sites:
  - https://github.com/Automattic/custom-fonts/blob/e9eb1ffe861baa5cd2c107d88338a23501aa82c6/custom-fonts.php#L97-L98
  - https://github.com/Automattic/custom-fonts-typekit/blob/0be6c997f3c576fc86844b5d57130d6ed81fe624/custom-fonts-typekit.php#L50
  - https://github.com/Automattic/wpcomsh/blob/2228359875ed6f13ae1164358457971dcdaa863d/wpcomsh.php#L76-L77

**Comment by mmtr** [comment]
> Jetpack fonts (x2 calls) - unsure what's going on - I don't get fonts in customizer with TT3 at least -

I took a further look at this and learnt that:
- Custom fonts controls are registered in the Customizer [if the theme has defined font rules](https://github.com/Automattic/custom-fonts/blob/e9eb1ffe861baa5cd2c107d88338a23501aa82c6/custom-fonts.php#L141).
- Font rules are defined by themes with [the `jetpack_fonts_rules` action](https://github.com/Automattic/custom-fonts/blob/e9eb1ffe861baa5cd2c107d88338a23501aa82c6/css-generator.php#L132).
- Seems that only a retired premium plan uses that: fbhepr%2Skers%2Sjcpbz%2Qcerzvhz%2Qgurzrf%2Szrybql%2Svap%2Ssbagf.cuc%3Se%3Qs7qq03o9%231058-og

**Comment by mmtr** [comment]
Tried today to add some `wp_is_block_theme` checks to all the actions identified above for WP.com Simple sites (except the footer credit, which we know needs extra work), since it seemed something trivial to do.

In theory, if we do something like this in all the places that register a `customize_register` action, we will prevent the Customize menu from showing up.

```php
if ( ! wp_is_block_theme() ) {
  add_action( 'customize_register', ... );
}
```

Turns out it is not that trivial. The problem is that Calypso, in order to render the menu, makes a request to the `admin-menu` endpoint and when the actions above are registered, `wp_is_block_theme` still points to the theme of the public API site, causing the "is block theme" checks to return wrong information.

Not sure why is that, since the public API definitely makes a `switch_to_blog` at some point but probably not early enough. So I think we might need to change something in the API infrastructure for all of this to work as expected.

---

In the process of investigating this, I also found that [the custom CSS nudge mentioned above for Atomic free sites](https://github.com/Automattic/wp-calypso/issues/78938#issuecomment-1638098901) also affect Simple free sites:
- https://github.com/Automattic/jetpack/blob/3fb51552fc8215fbe87cca93499e1415078fb5dc/projects/plugins/jetpack/modules/masterbar/nudges/bootstrap.php#L55
- fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Sphfgbz%2Qpff%2Qva%2Qphfgbzvmre.cuc%3Se%3Qs3rsp4po%23306-og
